Services/Init/classes/class.ilErrorHandling.php

```diff
@@ -410,13 +410,37 @@ protected function devmodeHandler() {
 			case "PLAIN_TEXT":
 				return new ilPlainTextHandler();
 			case "PRETTY_PAGE":
-				return new PrettyPageHandler();
+				// fallthrough
 			default:
-				if ($ilLog) {
-					$ilLog->write("Unknown or undefined error handler '".ERROR_HANDLER."'. "
-								 ."Falling back to PrettyPageHandler.");
+				if ((!defined('ERROR_HANDLER') || ERROR_HANDLER != 'PRETTY_PAGE') && $ilLog) {
+					$ilLog->write(
+						"Unknown or undefined error handler '".ERROR_HANDLER."'. " .
+						"Falling back to PrettyPageHandler."
+					);
+				}
+
+				$prettyPageHandler = new PrettyPageHandler();
+				if (defined('ERROR_EDITOR_URL')) {
+					$pathTranslations = [];
+
+					if (defined('ERROR_EDITOR_PATH_TRANSLATIONS')) {
+						$mappings = explode('|', ERROR_EDITOR_PATH_TRANSLATIONS);
+						foreach ($mappings as $mapping) {
+							$parts = explode(',', $mapping);
+							$pathTranslations[$parts[0]] = $parts[1];
+						}
+					}
```

Comment:
Do I understand correctly that this maps server paths to local paths? I would configure this as "server1,local1|server2,local2", correct? This is really brittle regarding spaces, right? I would at least slash a trim around the `$parts`. I also would suggest to factor this block out in a method called `getPathMappings` or something similar. This was hard to understand at first.

Reply:
Yes, exactly.

I will changed this,